### PR TITLE
tests: support for ubuntu core 22 in arm architecture

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes the bits needed to run ubuntu-core in arm devices.
Changes for nested tests are implemented but not enabled as kvm still is not working in arm servers.

Changes in tests to support the new system are also included.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -400,6 +400,7 @@ jobs:
         - ubuntu-core-18-64
         - ubuntu-core-20-64
         - ubuntu-core-22-64
+        - ubuntu-core-22-arm-64
```
Comment (alfonsosanchezbeato) [inline .github/workflows/test.yaml:403]: I'd prefer arm64 to match the debian architecture. I think that we are in general more used to that.

File: spread.yaml [hunk 2]
```diff
@@ -199,6 +199,10 @@ backends:
                   image: ubuntu-os-cloud/ubuntu-2204-lts-arm64
                   workers: 8
                   storage: 12G
+            - ubuntu-core-22-arm-64:
+                  image: ubuntu-22.04-arm-64
```
Comment (alfonsosanchezbeato) [inline spread.yaml:203]: Same about arm64

File: spread.yaml [hunk 3]
```diff
@@ -240,6 +244,18 @@ backends:
                   storage: 25G
                   workers: 8
 
+    google-nested-arm:
+        type: google
+        key: '$(HOST: echo "$SPREAD_GOOGLE_KEY")'
+        location: snapd-spread/us-central1-a
+        plan: t2a-standard-2
+        halt-timeout: 2h
+        cpu-family: "Intel Cascade Lake"
```
Comment (alfonsosanchezbeato) [inline spread.yaml:253]: Intel?

File: tests/lib/nested.sh [hunk 4]
```diff
@@ -1040,7 +1043,12 @@ nested_start_core_vm_unit() {
 
     local PARAM_MACHINE
     if [[ "$SPREAD_BACKEND" = google-nested* ]]; then
-        PARAM_MACHINE="-machine ubuntu${ATTR_KVM}"
+        if os.query is-arm; then
+            PARAM_MACHINE="-machine virt${ATTR_KVM}"
+            PARAM_CPU="-cpu cortex-a57"
```
Comment (alfonsosanchezbeato) [inline tests/lib/nested.sh:1048]: Maybe we should use "host", see https://qemu-project.gitlab.io/qemu/system/arm/cpu-features.html#a-note-about-cpu-models-and-kvm

File: .github/workflows/test.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply .github/workflows/test.yaml:403]: Already discussed. It will remain like this to match systems in other backends

File: spread.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply spread.yaml:203]: same

File: tests/lib/nested.sh (hunk 4 above)
Comment (sergiocazzolato) [reply tests/lib/nested.sh:1048]: Yesn thanks for the suggestion. Updated

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks for the changes

**Review by valentindavid — APPROVED** [review]
Looks good.

**Pull request merged**